packages/ember-views/lib/streams/key_stream.js

```diff
@@ -38,6 +38,10 @@ merge(KeyStream.prototype, {
     if (nextObj !== prevObj) {
       if (prevObj && typeof prevObj === 'object') {
         removeObserver(prevObj, this.key, this, this.notify);
+
+        if (nextObj && typeof nextObj === 'object') {
```

Comment:
I had that originally but it causes this assertion to fail: https://github.com/emberjs/ember.js/blob/master/packages/ember-views/tests/views/view/class_name_bindings_test.js#L214